addons/website/static/src/snippets/s_website_form/000.js

```diff
@@ -576,9 +576,9 @@ odoo.define('website.s_website_form', function (require) {
         _compareTo(comparator, value = '', comparable, between) {
             switch (comparator) {
                 case 'contains':
-                    return value.includes(comparable);
+                    return value && value.includes(comparable);
                 case '!contains':
-                    return !value.includes(comparable);
+                    return value && !value.includes(comparable);
```

Comment:
If having an empty value is a situation that can happen, can't it happen in the `value.includes(':')` below as well ?

Reply:
Well seen.

The issue is actually that value can be `null` when the field compared is not empty and supposed to be visible, but it's not yet retrievable from the FormData() at
* https://github.com/odoo/odoo/blob/00600ab9a7bfc97c5c4c558b36124625e65ec1ab/addons/website/static/src/snippets/s_website_form/000.js#L654-L655

Reply:
As I tried to rebase before pushing my modifications, I found that @qsm-odoo made a PR last week about that:
https://github.com/odoo/odoo/commit/2714469

Which is also why you couldn't reproduce scenario C.